src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UnitTest/UnitTestsControl.cs

```diff
@@ -247,165 +298,192 @@ private async Task RunTests(CancellationToken ct, string[] filters)
 
 				_ = ReportMessage("Running tests...");
 
-				foreach (var type in testTypes.Where(t => t.type != null))
+				foreach (var type in testTypes.Where(t => t.Type != null))
 				{
-					var tests = type.tests
-						.Where(t => (filters?.None() ?? true)
-						            || filters.Any(f => t.DeclaringType.FullName.Contains(f, StrComp))
-						            || filters.Any(f => t.Name.Contains(f, StrComp)))
-						.ToArray();
+					var tests = FilterTests(type, filters);
 
 					if (tests.Length == 0)
 					{
 						continue;
 					}
 
-					ReportTestClass(type.type.GetTypeInfo());
+					ReportTestClass(type.Type.GetTypeInfo());
 					_ = ReportMessage($"Running {tests.Length} test methods");
 
-					var instance = Activator.CreateInstance(type: type.type);
+					var instance = Activator.CreateInstance(type: type.Type);
+
+					await ExecuteTestsForInstance(ct, instance, counters, tests, type);
+				}
+
+				_ = ReportMessage("Tests finished running.", isRunning: false);
+				ReportTestsResults(counters);
+			}
+			catch (Exception e)
+			{
+				counters.Failed = -1;
+				_ = ReportMessage($"Tests runner failed {e}");
+				ReportTestResult("Runtime exception", TestResult.Failed, counters, e);
+				ReportTestsResults(counters);
+			}
+		}
+
+		private MethodInfo[] FilterTests(UnitTestClassInfo testClassInfo, string[] filters)
+		{
+			if (testClassInfo.Tests is null)
+			{
+				return Array.Empty<MethodInfo>();
+			}
+
+			var testClassNameContainsFilters = filters.Any(f => testClassInfo.Type.FullName.Contains(f, StrComp));
```

Comment:
You can inline this into the where-clause or change this into a `filters?.Any(...) ?? false`.
And, you dont have to worry about having to null-coalesce with `Array.Empty<string>()` every time.